datadictionary/datadictionary.go

```diff
@@ -305,15 +306,21 @@ func Parse(path string) (*DataDictionary, error) {
 	}
 	defer xmlFile.Close()
 
+	return ParseSrc(xmlFile)
+}
+
+//ParseSrc loads and and build a datadictionary instance from an xml source.
+func ParseSrc(xmlSrc io.Reader) (*DataDictionary, error) {
 	doc := new(XMLDoc)
-	decoder := xml.NewDecoder(xmlFile)
+	decoder := xml.NewDecoder(xmlSrc)
 	if err := decoder.Decode(doc); err != nil {
 		return nil, err
 	}
 
 	b := new(builder)
 	var dict *DataDictionary
-	if dict, err = b.build(doc); err != nil {
+	dict, err := b.build(doc)
```

Comment:
with this change, the declaration of dict above is no longer necessary

Reply:
ye, I thought it was there for stylistic reasons, fixing...